src/datasources/db/v2/database-migrator.service.ts

```diff
@@ -0,0 +1,157 @@
+import { PostgresDatabaseService } from '@/datasources/db/v2/postgres-database.service';
+import {
+  LoggingService,
+  type ILoggingService,
+} from '@/logging/logging.interface';
+import { Inject, Injectable } from '@nestjs/common';
+import { ConfigService } from '@nestjs/config';
+import type { DataSource } from 'typeorm';
+
+interface LockSchema {
+  id: number;
+  name: number;
+}
+
+enum migrationStatus {
+  RUNNING = 1,
+}
+
+@Injectable()
+export class DatabaseMigrator {
+  public constructor(
+    @Inject(LoggingService) private readonly loggingService: ILoggingService,
+
+    @Inject(PostgresDatabaseService)
+    private readonly databaseService: PostgresDatabaseService,
+
+    @Inject(ConfigService)
+    private readonly configService: ConfigService,
+  ) {
+    //
+  }
+
+  /**
+   * Performs the migration operation.
+   *
+   * @returns {Promise<void>} A promise that resolves when the migration is complete.
+   *
+   * @throws {Error} Throws an error if unsuccessful after a specified number of retries."
+   */
+  public async migrate(): Promise<void> {
+    this.loggingService.info('Migrations: Running...');
+
+    const connection =
+      await this.databaseService.initializeDatabaseConnection();
+    await this.createLockTableIfNotExists(connection);
+
+    let numberOfIterations = 0;
+    const numberOfRetries = await this.configService.getOrThrow(
+      'db.migrator.numberOfRetries',
+    );
+    while (numberOfRetries >= numberOfIterations) {
+      ++numberOfIterations;
+      if (!(await this.lockExists(connection))) {
+        await this.insertLock(connection);
+        await this.runMigration(connection);
+        await this.truncateLocks(connection);
```

Comment:
This is intentional. Ideally, errors shouldn't occur, but I wanted to lock the table temporarily in case of an error. This way, a developer will need to manually unlock the table after resolving the issue.I believe it's more reliable to unlock the table manually in the event of an error.

Reply:
It makes sense to me, thanks for explaining!